Automattic/wp-calypso
# User Licensing: Skip plans page during Jetpack connection

#### Changes proposed in this Pull Request

When connecting Jetpack, if the user has an available unattached product license key, in this PR we skip the "Plans" page and redirect back to the wp-admin Jetpack dashboard instead. Otherwise if there are no available licenses, we continue the default behavior of taking the user to the "Plans" page (`/jetpack/connect/plans/:site`).

**Note:** The Jetpack "_In-Place_" Connection is no longer in use and has been disabled since Jetpack 10.1, therefore the Jetpack plugin authorization & connection flow always takes place in Calypso at this time.

#### Testing instructions

Prerequisites:
- You'll need a 'detached' user license.  To create a user license:you can, 1. Log into your wpcom sandbox and create one using wp-shell(`wpsh`), instructions here: D68384-code. Or 2. you can enable a flag(`add_filter( 'jetpack_license_checkout_m2_enabled', '__return_true' );`) in your sandbox and complete checkout with a jetpack product, instructions here: D67941-code).

Steps:
1. Checkout this branch and `yarn start`.
1. Make sure you're logged-in to Calypso with the account that owns a "detached" license (See Prerequisites above)
1. Create a JN site, with the Snippets plugin included. (select 'More options' and include the "Code Snippets" plugin)
1. Add a new snippet with code: `define( 'CALYPSO_ENV', 'development' );`. **Save** the snippet and be sure to click "**Activate**".
1. Go to: the Jetpack plugin Dashboard page, `/wp-admin/?page=jetpack#/dashboard`
1. Click "Set up Jetpack" button.
1. Verify you are redirected to Calypso(development) for the authorization & connection process.
1. Click "Authorize"
1. Upon successful connection, verify you are redirected back to your wp-admin Jetpack Dashboard.

Now, test that a user account without a detached license, redirects to the `/jetpack/connect/plans` page upon successful connection.
1. In the plugin Jetpack Dashboard page, scroll down to "Connections", and click "Manage site connection", then "Disconnect"
1. In Calypso(development), log out the current account (that owns a detached license).
1. Then log-in to a different account that does **not** own a detached license.
1. Back to the plugin, refresh the page and click Set up Jetpack" button, Then "Authorize"
1. Upon successful connection, verify you are redirected `/jetpack/connect/plans` page.

**Run unit tests:**
 - `yarn test-client client/jetpack-connect/test/authorize.js`
 - Verify all tests pass.


Related to 1201096622142517-as-1201099910081851
